utils/bids/run_level.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+"""Determine level at which the gear is running."""
+
+import logging
+
+from flywheel import ApiException
+
+log = logging.getLogger(__name__)
+
+
+def get_run_level_and_hierarchy(fw, destination_id):
+    """
+    Determine the level at which a job is running, given a destination
+    To get the destination, use gtk_context.destination["id"]
+    Args:
+        fw (gear_toolkit.GearToolkitContext.client): flywheel client
+        destination_id (id): id of the destination of the gear
+
+    Returns:
+        hierarchy (dict): containing the run_level and labels for the
+            run_label, group, project, subject, session, and
+            acquisition.
+
+    Note:
+        This returns "no_parent" for run_level when the destination has no
+        parent and returns run_level of "no_destination" if the destination
+        has no "parents" attribute.  In both of these cases, if the job's
+        destination.type is "acquisition" then the job is likely a utility
+        gear running on an acquisition.
+        This function is meant to be run by analysis gears.
+    """
+
+    try:
```

Comment:
There are a few improvements that could happen here, I'd be happy to help optimize this with you, but since it's only run once per gear run, it's probably not super important to optimize.